lib/fluent/plugin/out_kinesis.rb

```diff
@@ -303,5 +303,16 @@ def scaling_factor
     def record_exceeds_max_size?(record_string)
       return record_string.length > PUT_RECORD_MAX_DATA_SIZE
     end
+
+    def build_sizeexceeded_error_message(record)
+      notification = '...(message truncated.)'
+      message = sprintf('Record exceeds the %.3f KB(s) per-record size limit and will not be delivered: %s', PUT_RECORD_MAX_DATA_SIZE / 1024.0, record)
+      if(message.length <= PUT_RECORD_MAX_DATA_SIZE)
+        message
+      else
+        truncated_message = message.slice(0,PUT_RECORD_MAX_DATA_SIZE - notification.length) + notification
```

Comment:
This logic is not enough because `log.error` record will be like:

```
{"message":"'+message+'","time":"2015-10-28T13:40:07Z","tag":"fluent.error"}
```

It means that more truncation is required.